src/ada/src/services/arv/automation_request_validator.ads

```diff
@@ -0,0 +1,277 @@
+      when Other_Task =>
+         null;
+      end case;
+   end record;
+
+   package Int64_Task_Maps is new Ada.Containers.Functional_Maps
+     (Key_Type     => Int64,
+      Element_Type => Task_Kind_And_Id);
+   type Task_Map is new Int64_Task_Maps.Map;
+
+   type Automation_Request_Validator_Configuration_Data is record
+      Available_Configuration_Entity_Ids : Int64_Set;
+      Available_State_Entity_Ids         : Int64_Set;
+      Available_KeepIn_Zones_Ids         : Int64_Set;
+      Available_KeepOut_Zones_Ids        : Int64_Set;
+      Available_Area_of_Interest_Ids     : Int64_Set;
+      Available_Line_of_Interest_Ids     : Int64_Set;
+      Available_Point_of_Interest_Ids    : Int64_Set;
+      Available_Operating_Regions        : Operating_Region_Map;
+      Available_Tasks                    : Task_Map;
+      Available_Initialized_Tasks        : Int64_Set;
+   end record;
+
+   --  The mutable state of the service is stored inside formal containers which
+   --  can be modified.
+
+   type Automation_Request_Type is
+     (Automation_Request,
+      Sandbox_Automation_Request,
+      Task_Automation_Request);
+
+   type Request_Details (requestType : Automation_Request_Type := Automation_Request) is record
+      case requestType is
+         when Sandbox_Automation_Request =>
+            Play_Id         : Int64 := 0;
+            Soln_Id         : Int64 := 0;
+         when Task_Automation_Request =>
+            Task_Request_Id : Int64 := 0;
+         when others =>
+            null;
+      end case;
+   end record;
+
+   package Int64_Request_Details_Maps is new Ada.Containers.Formal_Hashed_Maps
+     (Key_Type     => Int64,
+      Element_Type => Request_Details,
+      Hash         => Int64_Hash);
+
+   Request_Details_Max_Capacity : constant := 200; -- arbitrary
+   subtype Request_Details_Map is Int64_Request_Details_Maps.Map
+     (Request_Details_Max_Capacity,
+      Int64_Request_Details_Maps.Default_Modulus (Request_Details_Max_Capacity));
+
+   package UniqueAutomationRequest_Lists is new Ada.Containers.Formal_Doubly_Linked_Lists
+     (Element_Type => UniqueAutomationRequest);
+
+   Max_UniqueAutomationRequest_Deque_Depth : constant := 200; -- arbitrary
+   subtype UniqueAutomationRequest_Ref_Deque is UniqueAutomationRequest_Lists.List
+     (Capacity => Max_UniqueAutomationRequest_Deque_Depth);
+   use UniqueAutomationRequest_Lists;
+
+   type Automation_Request_Validator_State is record
+      Sandbox                    : Request_Details_Map;
+      Pending_Requests           : UniqueAutomationRequest_Ref_Deque;
+      Requests_Waiting_For_Tasks : UniqueAutomationRequest_Ref_Deque;
+   end record;
+
+   --------------------------------------
+   -- Functions for annotation purpose --
+   --------------------------------------
+
+   --  Those are not changed. The functionality of the service remains the
+   --  same.
+
+   function All_Elements_In (V : Int64_Seq; S : Int64_Set) return Boolean is
```

Comment:
These expression functions that are used for proof should be completed in the package private part, otherwise in the package body. They make the code too hard to read as-is.